### PR TITLE
Add logic to search/fetch physical objects

### PR DESCRIPTION
**Pull request by johnbradley**
Adds search_objects() function to search physical objects. The return value follows the pattern of the search_media() returned value.

Adds get_object() to fetch a PhysicalObject given a physical object id.

### CODE REVIEW

File: src/morphosource/search.py [hunk 1]
```diff
@@ -17,18 +17,47 @@ def _get(obj, name, unlist=True):
...
+        self.taxonomy = _get(data, "taxonomy")
+        self.data = data
+
+    def get_media_ary(self, open_visibility_only=False):
+        results = []
+        media_search_results = search_media(query=self.id)
+        for media in media_search_results.items:
+            if self.id == media.physical_object_id:
+                if open_visibility_only:
+                    # NOTE: The value returned in visiblity is different than the facet
```
Comment (egrace479) [inline src/morphosource/search.py:52]: What does this mean?

File: tests/test_search.py [hunk 2]
```diff
@@ -1,17 +1,26 @@
 import unittest
 import requests
 from unittest.mock import patch, Mock
-from morphosource.search import search_media, get_media, Endpoints, ItemNotFound
+from morphosource.search import search_media, get_media, Endpoints, ItemNotFound, \
```
Comment (egrace479) [inline tests/test_search.py:4]:
```suggestion
from morphosource.search import search_media, get_media, Endpoints, ItemNotFound, 
```

File: src/morphosource/search.py (hunk 1 above)
Comment (johnbradley) [reply src/morphosource/search.py:52]: In the MorphoSource UI there is a `Publication` facet that can be filtered by the values `Open Download` and `Restricted Download`. The JSON payload returned from the MorphoSource API includes a `visibility` field that contains the same data but uses different values: `open` and `restricted_download`. So the `if` below filters for "open".

Comment (johnbradley) [reply src/morphosource/search.py:52]: This is for open access downloads. After a second thought I could move the filtering further up and use the facet.

File: tests/test_search.py (hunk 2 above)
Comment (johnbradley) [reply tests/test_search.py:4]: I get an error on the next line if I remove the backslash.

**Comment by egrace479** [comment]
I'm getting an error in `test_search.py`, specifically `test_physical_object_get_media_ary` at line 210:
```
AssertionError: expected call not found.
Expected: search_media(query='000577960', visibility=None)
Actual: search_media(query='000577960')
```

**Comment by johnbradley** [comment]
@egrace479 I didn't get this error when running the tests.
Could you try running this command within your virtual environment to install the latest version:
```
pip uninstall morphosource -y && pip install .
```
Then re-run the tests.

**Review by egrace479 — APPROVED** [review]
LGTM

**Review by thompsonmj — APPROVED** [review]
Great!

**Pull request merged**